machine_learning_hep/analysis/analyzer_jet.py

```diff
@@ -3082,53 +3242,149 @@ def jetsystematics(self):
             tg_powheg.append(tg_sys(input_powheg_z[ibin2], input_powheg_sys_z[ibin2]))
             tg_powheg_xsection.append(tg_sys(input_powheg_xsection_z[ibin2], input_powheg_xsection_sys_z[ibin2]))
 
+        powheg_ratio_sys = []
+        powheg_ratio = []
+        file_sys_out.cd()
+        for ibin2 in range(self.p_nbin2_gen):
+            input_powheg_z[ibin2].Write()
+            tg_powheg[ibin2].Write()
+            name_powheg = input_powheg_z[ibin2].GetName()
+            name_tg = tg_powheg[ibin2].GetName()
+            if self.lc_d0_ratio:
+                input_powheg_lc = input_file_lc.Get(name_powheg)
+                input_powheg_tg_lc = input_file_lc.Get(name_tg)
+                input_powheg_ratio = input_powheg_lc.Clone("input_powheg_ratio")
+                input_powheg_ratio.Divide(input_powheg_z[ibin2])
+                shapebins_error=[]
+                shapebins_contents=[]
+                shapebins_centres=[]
+                shapebins_widths=[]
+                for ibinshape in range(self.p_nbinshape_gen):
+                    if input_powheg_lc.GetBinContent(ibinshape+1)!=0:
+                        tg_error_lc = input_powheg_tg_lc.GetErrorY(ibinshape)/input_powheg_lc.GetBinContent(ibinshape+1)
+                        tg_error = tg_powheg[ibin2].GetErrorY(ibinshape)/input_powheg_z[ibin2].GetBinContent(ibinshape+1)
+                    else:
+                        tg_error_lc = 0
+                        tg_error = 0
+                    abs_err = sqrt(tg_error_lc*tg_error_lc + tg_error*tg_error)*input_powheg_ratio.GetBinContent(ibinshape+1)
+                    shapebins_error.append(abs_err*0.5)
```

Comment:
this should be checked but it doesnt look like it should be halved

Reply:
Yes, there is a problem. Fixed.